src/kernel/mm.c

```diff
@@ -0,0 +1,47 @@
+#include "mm.h"
+#include "panic.h"
+
+void initMm() {
+    struct KmallocHeader *chunk = (struct KmallocHeader*)MEMORY_START;
+    chunk->size = TOTAL_MEMORY;
+    chunk->user = MEMORY_OWNER_FREE;
+}
+
+void *kmalloc(u16 owner, u16 size) {
+    struct KmallocHeader *chunk, *other;
+    u16 realSize;
+
+    if ((realSize = sizeof(struct KmallocHeader) + size) < MEMORY_MIN_SIZE)
+        realSize = MEMORY_MIN_SIZE;
+    chunk = (struct KmallocHeader*)MEMORY_START;
+    while (chunk->user != MEMORY_OWNER_FREE || chunk->size < realSize) {
```

Comment:
What if all the blocks are used?
You could add a 0 sized block in the init, it would be another stopping condition.